internal/controller/drplacementcontrol_controller.go

```diff
@@ -160,11 +163,15 @@ func (r *DRPlacementControlReconciler) Reconcile(ctx context.Context, req ctrl.R
 
 	placementObj, err = getPlacementOrPlacementRule(ctx, r.Client, drpc, logger)
 	if err != nil && !(k8serrors.IsNotFound(err) && rmnutil.ResourceIsDeleted(drpc)) {
+		isMainDrpcRetrieved = false
```

Comment:
I believe a global scope variable here would interfere with other DRPC reconciliation routines that are running. I.e a DRPC that is being reconciled will set the global and then the other DRPC reconciliation will pick up that value.

A per DRPC instance variable would be required.